metrics/stackdriver_exporter.go

```diff
@@ -63,10 +75,32 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGCPMetadata(config)
 	mtf := getMetricTypeFunc(config.stackdriverMetricTypePrefix, config.stackdriverCustomMetricTypePrefix)
+
+	proj := config.stackdriverProjectID
+	if config.stackdriverConfig.ProjectID != "" {
+		proj = config.stackdriverConfig.ProjectID
+	}
+
+	var co []option.ClientOption
+	if config.stackdriverConfig.GCPSecretName != "" {
+		secret, err := getStackdriverSecretFunc(&config.stackdriverConfig)
+
+		if err == nil {
+			co = []option.ClientOption{sdconfig.ConvertSecretToExporterOption(secret)}
+		} else {
+			logger.Errorf("Failed to retrieve Secret for authenticating with Stackdriver: ", zap.Error(err))
+			// Automatically fall back on Google application default credentials
+		}
+	}
```

Comment:
Could we move this part to `getGCPMetadata` as well?

Reply:
This is more using GCP metadata than getting it. It's also a bit weird to move it into getGCPMetadata because it interacts with Kubernetes. I will move it into its own getExporterClientOption helper function though.